fastapi/params.py

```diff
@@ -379,3 +380,16 @@ def __init__(
     ):
         super().__init__(dependency=dependency, use_cache=use_cache)
         self.scopes = scopes or []
+
+
+class DependsContext:
+    def __init__(self, param: Parameter) -> None:
+        self.param = param
```

Comment:
I'm wrapping the single item in a custom class so that in the future we can add more stuff here in a backwards compatible manner. E.g. I can imagine a world where a dependency wants to know what path it is registered under, what dependencies depend on it, etc.